app/jobs/import_from_vacancy_source_job.rb

```diff
@@ -4,23 +4,24 @@ class ImportFromVacancySourceJob < ApplicationJob
   def perform(source_klass)
     return if DisableExpensiveJobs.enabled?
 
+    @source_klass = source_klass
     @source_name = source_klass.source_name
-    import_from_source(source_klass)
+    @vacancies_count = 0
+    @external_references = []
+    @errors = {}
+
+    import_from_source
+    report_validation_errors
+    mark_removed_vacancies_from_source
```

Comment:
This is a refactor to make `import_from_source` more readable (or at least I think so) and `perform` commands the steps one by one. I find this easier to follow now, but let me know if you think otherwise!